### DISCUSSION TITLE
dnsdist checking the presence of a record on the backend

### DISCUSSION THREAD

**Discussion by deadsik** [PowerDNS/pdns#13829] [General]
Hello,

Please tell me about balancing, is there a parameter that would check the presence of an entry on the backend and if there is an entry, give it, if there is no entry, check another backend?

I have several backends and different zones on them, some on one and some on the other.
```
setACL({
        "0.0.0.0/0",
        "::/0"
})

addLocal("0.0.0.0:53")
addLocal("[::]:53")

setServerPolicy(whashed)
newServer({address="1.1.1.1:53", pool="default", name="serv1"})
newServer({address="2.2.2.2:53", pool="default", name="serv2"})

addAction(AllRule(), PoolAction("default"))
```

*for example, I now have this config, backends  1.1.1.1, 2.2.2.2 (example). On backend: serv1 i have zone: example1.com, on backend serv2 i have zone: example2.com.  As I understand it, the records are taken randomly from them.

**Comment by deadsik** [comment]
I know how to make a specific backend for a specific zone in the config, but if there are many zones, is such balancing possible so as not to register each zone separately.

**Comment by rgacogne** [comment]
I would strongly suggest maintaining a list of zones and routing on that zone. The list does not have to be static, dnsdist can for example query a CDB or LMDB database as documented in https://dnsdist.org/reference/kvs.html
If you really do not want to maintain a list, and your backend returns a specific code when it does not know about a zone (like for example `REFUSED`) then you can tell dnsdist to send the query again to a different pool. See https://dnsdist.org/reference/dq.html#DNSResponse:restart
